definitions/infra-container/definition.yml

```diff
@@ -3,25 +3,38 @@ type: CONTAINER
 
 synthesis:
   rules:
+  # Dockerstats receiver from otel collector.
   - identifier: container.id
     name: container.name
     encodeIdentifierInGUID: true
-    # This is a temporary condition to restrict synthesis for a particular set of telemetry data.
     conditions:
-      - attribute: nr.entity_type
-        value: container
+      - attribute: eventType
+        value: MetricRaw
+      # All metrics from the receiver starts with the 'container.' prefix.
+      - attribute: metricName
+        prefix: container.
+      - attribute: instrumentation.provider
+        value: opentelemetry
```

Comment:
I'm not sure if the attribute used for this should be `newrelic.source`, I've seen different approaches for this: [host](https://github.com/newrelic/entity-definitions/blob/main/definitions/infra-host/definition.yml#L33), [redis](https://github.com/newrelic/entity-definitions/blob/d5a26a14675b5c532ecbb3fcb274fd62396819c2/definitions/ext-redis/definition.yml#L28)

Reply:
I think either `instrumentation.provider` or `newrelic.source` will work. `newrelic.source` allows to filter to a specific pipeline, in this case the OTLP metric pipeline, while `instrumentation.provider` will apply to all OTLP pipelines (Metrics, Traces, Logs)